src/bindings/g_object/object.cr

```diff
@@ -508,6 +508,12 @@ module GObject
         end
 
         def emit({{ signature.args.splat }})
+          {% for arg in signature.args %}
+            {% resolved_type = arg.type.resolve %}
+            {% if resolved_type < Reference && !resolved_type.has_method?(:to_unsafe) %}
```

Comment:
You still check if the `to_unsafe` method exists.
What if I want to pass an `XML::Node`?
It also has a `to_unsafe` method, but that returns the underlying C struct from LibXML, which is hard to convert to a crystal object again.

Reply:
You are right.. I need to check `resolved_type < Reference && resolved_type > GObject::Object`.

Reply:
in fact a bit more complex than that... since there's the String use case too.